### PR TITLE
RHCLOUD-36016: Add workspaces integration test to source tree

### PR DESCRIPTION
**Pull request by catastrophe-brandon**
Highlights:
* Uses Mockoon for service stubbing (no coding needed for the service stubs, entirely generated from the API spec). Produces API data automagically.
* Tests were tweaked to focus on individual workspaces endpoints and less on the CRUD workflow. They could probably benefit from some additional updates with assertions, but it's a good starting point.
* a bunch of helper code around auth was removed bc it wasn't needed
* These tests would fall under "integration tests" rather than "e2e" but I think there's plenty of benefit from the perspective of validating client request-response workflow.
* This update only covers ONE API. There are still details that would need to be ironed-out if we expand to cover multiple APIs. (But nothing we couldn't find a solution for)
* No need for ephemeral, stage, SSO, etc. Self-contained.

### CODE REVIEW

File: .github/actions/test-e2e/action.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: E2E tests
```
Comment (catastrophe-brandon) [inline .github/actions/test-e2e/action.yml:1]: Probably rename this to "Integration tests". Technically this isn't "e2e" with a fully built-out back-end.

File: package.json [hunk 2]
```diff
@@ -43,6 +45,7 @@
   "homepage": "https://github.com/RedHatInsights/javascript-clients#readme",
   "dependencies": {
     "@emotion/react": "11.11.1",
+    "@mockoon/cli": "^9.0.0",
```
Comment (catastrophe-brandon) [inline package.json:48]: Not sure if this needs to move to dev dependencies?

File: packages/rbac/v2/tests/integration/client/index.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    WorkspacesPatch: WorkspacesPatch,
+    WorkspacesUpdate: WorkspacesUpdate,
+  },
+  { axios: axiosInstance, configuration: config },
+);
+
+// --- convenience wrappers for workspace operations ---
+
+export const createWorkspace = async (name: string, description: string, config?: AxiosRequestConfig) => {
+  // @ts-ignore
```
Comment (florkbr) [inline packages/rbac/v2/tests/integration/client/index.ts:39]: @catastrophe-brandon to follow up and move assertions to the tests themselves and remove ts-ignores (fix issues).

File: packages/rbac/v2/tests/integration/client/index.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import axios from 'axios';
+
+import { APIFactory } from '@redhat-cloud-services/javascript-clients-shared/utils';
+import { Configuration } from '@redhat-cloud-services/javascript-clients-shared/configuration';
+
+export const WORKSPACES_API_BASE = 'http://localhost:3000/api/rbac/v2';
+export const config: Configuration = new Configuration();
+const axiosInstance = axios.create();
+
+export const composedWorkspacesApi = APIFactory(
```
Comment (florkbr) [inline packages/rbac/v2/tests/integration/client/index.ts:23]: @apinkert @florkbr TODO - we should generate an API factory for V2 (really any sub version).

File: packages/rbac/v2/tests/integration/scripts/serve.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+echo "Running API using the updated spec"
+npx mockoon-cli start --data /tmp/workspaces_updated.json
```
Comment (florkbr) [inline packages/rbac/v2/tests/integration/scripts/serve.sh:4]: @catastrophe-brandon I think we can follow up to make this more reusable/put this script in a shared location outside of rbac.

**Review by florkbr — APPROVED** [review]
I think this is a great first pass at adding tests for our clients (without being flaky or intrusive). Super cool that the server and responses are generated based on the openAPI specs too. 

:shipit:

**Pull request merged**